EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java

```diff
@@ -1,257 +1,795 @@
+					result.addFailure(new ResponseException(500, NGSIConstants.ERROR_UNEXPECTED_RESULT,
+							NGSIConstants.ERROR_UNEXPECTED_RESULT_NULL_TITLE, statusCode, host, attribs));
+
+				} else {
+					if (!responseBody.containsKey(NGSIConstants.ERROR_TYPE)
+							|| !responseBody.containsKey(NGSIConstants.ERROR_TITLE)
+							|| !responseBody.containsKey(NGSIConstants.ERROR_DETAIL)) {
+						result.addFailure(
+								new ResponseException(statusCode, responseBody.getString(NGSIConstants.ERROR_TYPE),
+										responseBody.getString(NGSIConstants.ERROR_TITLE),
+										responseBody.getMap().get(NGSIConstants.ERROR_DETAIL), host, attribs));
+					} else {
+						result.addFailure(new ResponseException(500, NGSIConstants.ERROR_UNEXPECTED_RESULT,
+								NGSIConstants.ERROR_UNEXPECTED_RESULT_NOT_EXPECTED_BODY_TITLE, responseBody.getMap(),
+								host, attribs));
+					}
+				}
+			}
+		}
+		return Uni.createFrom().voidItem();
 	}
 
-	/**
-	 * Method to append fields in existing Entity in system/kafka topic
-	 * 
-	 * @param entityId - id of entity to be appended
-	 * @param resolved - jsonld message containing fileds to be appended
-	 * @return AppendResult
-	 * @throws ResponseException
-	 * @throws IOException
-	 */
-	public Uni<UpdateResult> appendToEntry(ArrayListMultimap<String, String> headers, String entityId,
-			Map<String, Object> resolved, String[] options, BatchInfo batchInfo) {
-		logger.trace("appendMessage() :: started");
-		// get message channel for ENTITY_APPEND topic
-		// payload validation
-		if (entityId == null) {
-			return Uni.createFrom()
-					.failure(new ResponseException(ErrorType.BadRequestData, "empty entity id is not allowed"));
+	public Uni<NGSILDOperationResult> partialUpdateAttribute(String tenant, String entityId, String attribName,
+			Map<String, Object> payload, Context originalContext) {
+		logger.trace("updateMessage() :: started");
+		Map<String, Object> effectivePayload;
+		if (payload.containsKey(attribName)) {
+			effectivePayload = payload;
+		} else {
+			effectivePayload = Maps.newHashMap();
+			effectivePayload.put(attribName, payload);
 		}
 
-		String tenantId = HttpUtils.getInternalTenant(headers);
-		// get entity details
-		return EntryCRUDService.validateIdAndGetBody(entityId, tenantId, entityInfoDAO).onItem()
-				.transform(Unchecked.function(t -> new AppendEntityRequest(headers, entityId, t, resolved, options)))
-				.onItem().transformToUni(t -> {
-					if (t.getUpdateResult().getUpdated().isEmpty()) {
-						return Uni.createFrom().item(t.getUpdateResult());
-					}
-					t.setBatchInfo(batchInfo);
-					return handleRequest(t).onItem().transform(v -> {
-						logger.trace("partialUpdateEntity() :: completed");
-						return t.getUpdateResult();
-					});
-				});
+		UpdateEntityRequest request = new UpdateEntityRequest(tenant, entityId, effectivePayload, attribName, null);
+		return entityDAO.partialUpdateAttribute(request).onItem().transformToUni(resultetEntity -> {
```

Comment:
Please verify partial update attributes functionality for entity.

Reply:
it's working.